modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2023 @@
+      paramError("energy_wall_types",
+                 "Size is not the same as the number of wall boundaries in 'wall_boundaries'");
+  }
+}
+
+void
+NSFVAction::checkAmbientConvectionParameterErrors()
+{
+  if (_has_energy_equation)
+  {
+    if (_ambient_convection_blocks.size() && _blocks.size())
+    {
+      for (auto const & block : _ambient_convection_blocks)
+        if (std::find(_blocks.begin(), _blocks.end(), block) == _blocks.end())
+          paramError("ambient_convection_blocks",
+                     "Block '" + block + "' is not present in the block IDs of the module!");
+
+      if (_ambient_convection_blocks.size() != _ambient_convection_alpha.size())
+        paramError("ambient_convection_alpha",
+                   "The number of heat exchange coefficients is not the same as the number of "
+                   "ambient convection blocks!");
+
+      if (_ambient_convection_blocks.size() != _ambient_temperature.size())
+        paramError("ambient_temperature",
+                   "The number of ambient temperatures is not the same as the number of "
+                   "ambient convection blocks!");
+    }
+    else if ((!_ambient_convection_blocks.size() && !_blocks.size()) ||
+             (!_ambient_convection_blocks.size() && _blocks.size()))
+    {
+      if (_ambient_convection_alpha.size() > 1)
+        paramError("ambient_convection_alpha",
+                   "The user should only use one or zero heat exchange coefficient if the ambient "
+                   "convection blocks are not defined!");
+      if (_ambient_convection_alpha.size() != _ambient_temperature.size())
+        paramError("ambient_temperature",
+                   "The number of ambient temperatures is not the same as the number of "
+                   "heat exchange coefficients!");
+    }
+    else if (_ambient_convection_blocks.size() && !_blocks.size())
+      paramError("ambient_convection_blocks",
+                 "If there are no subdomains defined in 'blocks', ambient convection blocks should "
+                 "not be defined either!");
+  }
+}
+
+void
+NSFVAction::checkFrictionParameterErrors()
+{
+  if (!_porous_medium_treatment)
+  {
+    if (isParamValid("friction_blocks"))
+      paramError("friction_blocks", "Friction is only supported for porous medium treatment!");
+    if (isParamValid("friction_types"))
+      paramError("friction_types", "Friction is only supported for porous medium treatment!");
+    if (isParamValid("friction_coeffs"))
+      paramError("friction_coeffs", "Friction is only supported for porous medium treatment!");
+  }
+
+  if (_friction_blocks.size() && _blocks.size())
+  {
+    for (auto const & block : _friction_blocks)
+      if (std::find(_blocks.begin(), _blocks.end(), block) == _blocks.end())
+        paramError("friction_blocks",
+                   "Block '" + block + "' is not present in the block IDs of the module!");
```

Comment:
I think it would be hard to do it since the types of the parameters are different. For example for ambient convection we have the ambient temperature as a vector<MooseFunctorName>, while for the friction term we have the coefficients as vector<vector<string>>. I tried to see if I could create a templated function for this purpose but then we need to pass names of the parameters, the corresponding vectors that need checking and so on. Also at the end I don't think it would look nicer and I think this improves flexibility for future extensions if we change the friction part but doesn't want to change the ambient part.

Reply:
I'll separate the block checks, but the block parameters should stay in my opinion.